astropy/io/fits/convenience.py

```diff
@@ -409,6 +414,103 @@ def writeto(filename, data, header=None, output_verify='exception',
                 checksum=checksum)
 
 
+def table_to_hdu(table):
+    """
+    Convert an astropy.table.Table object to a FITS BinTableHDU
+
+    Parameters
+    ----------
+    table : astropy.table.Table
+        The table to convert.
+
+    Returns
+    -------
+    table_hdu : astropy.io.fits.BinTableHDU
+        The FITS binary table HDU
+    """
+    # Avoid circular imports
+    from .connect import is_column_keyword, REMOVE_KEYWORDS
+
+    # Tables with mixin columns are not supported
+    if table.has_mixin_columns:
+        mixin_names = [name for name, col in table.columns.items()
+                       if not isinstance(col, table.ColumnClass)]
+        raise ValueError('cannot write table with mixin column(s) {0} to FITS'
```

Comment:
This exception message will seem a bit confusing if calling `table_to_hdu` directly (since at that point there is nothing about writing happening).  Maybe `"cannot create a FITS HDU from a table with mixin column(s) {}"`.  I think this message will also be more descriptive in the case of calling `t.write('out.fits')`.